[MRG] Fixes issue #6746 "number of power iterations" message

chriskaschner opened the pull request:
#### Reference Issue

Fixes issue #6746 
#### What does this implement/fix? Explain your changes.
- Changes default value passed to extmath.py from pca.py to None.
#6749 addressed the default within extmath.py and added some checking for whether the "number of power iterations" message was needed but didn't address the value being passed from pca.py.

chriskaschner committed:
Fixes issue #6746

- Changes default value passed to extmath.py from pca.py to None.

nelson-liu commented:
Can you update the docs as well to reflect the new defaults?

nelson-liu commented:
also i'm not sure that this is the way to go....might also want to check out http://scikit-learn.org/dev/developers/contributing.html#deprecation

amueller commented:
there are tests errors. And this shouldn't change any user behavior, apart from no warning.

amueller commented:
I'm pretty confused now. I would have left the explicit number here, and change the default to None in extmath.

chriskaschner committed:
Reference Issue
Fix issue #6746

What does this implement/fix? Explain your changes.
* Change default value passed to extmath.py from pca.py to None.
* update tests to reference the value of "None" being passed to extmath.py

chriskaschner committed:
Fix: fix test case for deprecated pca example

Fixes issue #6746

What does this implement/fix? Explain your changes.

Changes default value passed to extmath.py from pca.py to None.

chriskaschner committed:
Fix: Update Doctests to reflect new value 'iterated_power'

Fix issue #6746

What does this implement/fix? Explain your changes.

Changes default value passed to extmath.py from pca.py to None.
Update Docs and Doc Tests to reflect new value

amueller commented:
I still don't understand what's happening here... Shouldn't the changes be in the randomized svd function?

chriskaschner committed:
Fix: Update Doctests to reflect new value 'iterated_power'

Fix issue #6746

What does this implement/fix? Explain your changes.

Changes default value passed to extmath.py from pca.py to None.
Update Docs and Doc Tests in to reflect new value

chriskaschner commented:
@amueller the output from that function is referenced in a number of different doc tests, so the previously discussed changes required updates to the docs/ tests that referenced the default output of PCA.

amueller commented:
fixed by #7311.
